crates/router/src/core/payments.rs

```diff
@@ -6564,8 +6564,14 @@ pub async fn payment_external_authentication(
         &payment_attempt.clone(),
         payment_connector_name,
     ));
+    let merchant_connector_account_id = payment_attempt
+        .merchant_connector_id
```

Comment:
https://github.com/juspay/hyperswitch/pull/6954#discussion_r1900634441

The comment would also be applicable here. Same change (with respect to `create_webhook_url()`) had been taken up in a different PR